app/code/Magento/InventoryImportExport/Model/Export/Sources.php

```diff
@@ -0,0 +1,290 @@
+            $writer->writeRow($data);
+        }
+
+        return $writer->getContents();
+    }
+
+    /**
+     * @param Collection $collection
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    private function applyFilters(Collection $collection)
+    {
+        foreach ($this->retrieveFilterDataFromRequest() as $columnName => $value) {
+            $attributeDefinition = $this->getAttributeCollection()->getItemById($columnName);
+            $type = null;
+            if ($attributeDefinition) {
+                $type = $attributeDefinition->getData('backend_type');
+            }
+
+            $filterMethodName = 'applyFilterFor' . ucfirst($type);
+            if (method_exists($this, $filterMethodName)) {
+                $this->$filterMethodName($collection, $columnName, $value);
+            } else {
+                $this->applyDefaultFilter($collection, $columnName, $value);
+            }
+        }
+    }
+
+    /**
+     * @param Collection $collection
+     * @param string $columnName
+     * @param mixed $value
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function applyFilterForInt(Collection $collection, $columnName, $value)
+    {
+        if (is_array($value)) {
+            $from = $value[0] ?? null;
+            $to = $value[1] ?? null;
+
+            if (is_numeric($from) && !empty($from)) {
+                $collection->addFieldToFilter($columnName, ['from' => $from]);
+            }
+
+            if (is_numeric($to) && !empty($to)) {
+                $collection->addFieldToFilter($columnName, ['to' => $to]);
+            }
+
+            return;
+        }
+
+        $collection->addFieldToFilter($columnName, ['eq' => $value]);
+    }
+
+    /**
+     * @param Collection $collection
+     * @param string $columnName
+     * @param mixed $value
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function applyFilterForDecimal(Collection $collection, $columnName, $value)
+    {
+        $this->applyFilterForInt($collection, $columnName, $value);
+    }
+
+    /**
+     * @param Collection $collection
+     * @param string $columnName
+     * @param mixed $value
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function applyDefaultFilter(Collection $collection, $columnName, $value)
+    {
+        $collection->addFieldToFilter($columnName, ['like' => '%' . $value . '%']);
+    }
+
+    /**
+     * @param Collection $collection
+     * @param string $columnName
+     * @param mixed $value
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function applyFilterForDatetime(Collection $collection, $columnName, $value)
+    {
+        if (is_array($value)) {
+            $from = $value[0] ?? null;
+            $to = $value[1] ?? null;
+
+            if (is_scalar($from) && !empty($from)) {
+                $date = (new \DateTime($from))->format('m/d/Y');
```

Comment:
better not to use DateTime object directly 

but Magento\Framework\Stdlib\DateTime instead